### DISCUSSION TITLE
Can't find baro chip, what I have missed?

### DISCUSSION THREAD

**Discussion by lida2003** [iNavFlight/inav#9399] [Q&A] [answered]
Hi

I 'm using AocodaRC F405V2 board and substitue AT32F435. It works fine with atbetaflight, which suggest IMU/baro chip is all working.

Now, I have an issue "baro chip dps310 is not detected." There are a couple of items I have checked.

**a) It's a AocodaRC F405V2, so the basic configuration is from this target file.**
**b) I2C is the same ports  // should be correct**
```
#define I2C1_SCL                        PB6        // SCL
#define I2C1_SDA                        PB7        // SDA
```
**c) DPS310 chip is selected `USE_BARO_DPS310`**
**d) drivers for at32F435 has been built**
```
...
[  0%] Building C object src/main/target/AOCODARCF435V2/CMakeFiles/AOCODARCF435V2.elf.dir/__/__/drivers/bus_i2c_at32f43x.c.obj
...
[100%] Building C object src/main/target/AOCODARCF435V2/CMakeFiles/AOCODARCF435V2.elf.dir/__/__/drivers/barometer/barometer_dps310.c.obj
...
```
**e) configuration tab selects DPS310**
![图片](https://github.com/iNavFlight/inav/assets/1296171/f12a1700-0481-4fc9-b547-b7d3effe792d)

BUT, baro chip is not detected. what I have missed here?

Please check following configurations: [AOCODARCF435V2 ](https://github.com/SnapDragonfly/inav/tree/Copter-7.x-AocodaRC/src/main/target/AOCODARCF435V2)

**Comment by lida2003** [comment]
Now I have got betaflight 4.5 working.

```
# status
MCU AT32F435 Clock=288MHz, Vref=3.29V, Core temp=63degC
Stack size: 2048, Stack address: 0x2002fff0
Configuration: CONFIGURED, size: 4057, max available: 16384
Devices detected: SPI:1, I2C:1
Gyros detected: gyro 1 locked dma
GYRO=MPU6500, ACC=MPU6500, BARO=DPS310
OSD: MAX7456 (30 x 13)
System Uptime: 7 seconds, Current Time: 2023-10-24T06:02:28.061+00:00
CPU:20%, cycle time: 124, GYRO rate: 8064, RX rate: 15, System rate: 9
Voltage: 0 * 0.01V (0S battery - NOT PRESENT)
I2C Errors: 0
FLASH: JEDEC ID=0x00ef4018 16M
GPS: NOT ENABLED
Arming disable flags: RXLOSS ANGLE CLI MSP

# version
# Betaflight / AT32F435G (A435) 4.5.0 Oct 24 2023 / 13:58:02 (4f17fe0aa) MSP API: 1.46
# config rev: 6ab06df
# board: manufacturer_id: SJET, board_name: AOCODAF435V2MPU6500
```

![图片](https://github.com/iNavFlight/inav/assets/1296171/9bb2910d-dc48-4629-ac31-b8ef3bccb5f3)

**Comment by mmosca** [comment]
Let me know if you made any progress with the investigation.
Was the signaling looking ok under the osciloscope?

**Reply by lida2003** [reply]
[AT32F435+BMP280.zip](https://github.com/iNavFlight/inav/files/13163391/AT32F435%2BBMP280.zip)  // logic data in sal format

We have captured a frame since boot, which is NOT good.

![图片](https://github.com/iNavFlight/inav/assets/1296171/8e97a971-3c53-4039-886b-fe09eebbc929)

Test steps:

1)  flash bf4.5, test BMP280, if it's OK；
2)  flash inav fw，configure baro BMP280 in configuration tab；
3)  capture SCL/SDA wave during boot ( 3~4 seconds)

--- preliminary results:
a) As we had seen I2C wave on the bus. In my opinion, configuration for I2C is OK. 
b) And I have done serial debuging, which suggests dsp310/bmp280 driver code is called during boot up procedure.
c) Since BF4.5 works fine with BMP280/DPS310, I'm quite confident that the hardware is good to go.

so the most possible cause might be due to the driver or related issues. 

I think comparison between BF and iNav will be more effective way to find out the root cause.

**Reply by lida2003** [reply, marked as answer]
There is a comment on the issue:  ["i2C1 using ports PB6, PB7 not working for AT32F435 target. Need to hack/change "drivers/bus_i2c_at32f43x.c" for i2c1 to be mux4. In the code its mux8. "](https://github.com/iNavFlight/inav/pull/8752#issuecomment-1752145189)

Add a PR for this: https://github.com/iNavFlight/inav/pull/9412

![194c8a03024eab9debd2e4b17f062ab](https://github.com/iNavFlight/inav/assets/1296171/07831416-9ca8-4a8d-b040-266f6d9955da)
